user_interface.cpp

```diff
@@ -0,0 +1,112 @@
+#include <iostream>
+#include <string>
+#include "user_interface.h"
```

Comment:
For readability, put the header include on the first line separated with a newline (empty line) from the other includes.